### PR TITLE
Convert inserted tuples to the chunk's rowtype

### PR DESCRIPTION
**Pull request by erimatnor**
When routing inserted tuples to chunks, the rowtype of the chunk is
expected to match that of the root table. However, this is not always
the case if the root table has been modified, e.g., it had a column
removed. Newly created chunks will in that case have a different
rowtype and therefore tuples need to be converted to the chunk's
rowtype.

### CODE REVIEW

File: src/chunk_insert_state.c [hunk 1]
```diff
@@ -149,6 +150,19 @@ chunk_insert_state_create(Chunk *chunk, ChunkDispatch *dispatch)
 	if (parse != NULL && parse->onConflict != NULL)
 		state->arbiter_indexes = chunk_infer_arbiter_indexes(rti, dispatch->estate->es_range_table, dispatch->parse);
 
+	/* Set tuple conversion map, if tuple needs conversion */
+	parent_rel = heap_open(chunk_get_parent_oid(chunk->table_id), AccessShareLock);
```
Comment (cevian) [inline src/chunk_insert_state.c]: Shouldn't the parent here be dispatch->hypertable->main_table_relid. I think we can avoid a lookup here, no?

**Comment by erimatnor** [comment]
Yes, plan is to add tests.

**Comment by erimatnor** [comment]
Tests added. Addressed other comment.

File: test/sql/alter.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
+\ir include/create_single_db.sql
```
Comment (cevian) [inline test/sql/alter.sql]: Should we also have some test for dropping and recreating a column with the same name for both tests below? I imagine we want both one column that's just dropped and one that's dropped and recreated. Also do we want to make modifications do ddl_ops_1 and ddl_ops 2 to stress this?

**Comment by erimatnor** [comment]
Added tuple conversion for COPY path (which was left out initially) and expanded tests.

**Comment by mfreed** [comment]
Thanks to @ryan-shaw for discovering and assisting us with this issue.

**Pull request merged**